cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/OAuth2TokenAuthenticator.java

```diff
@@ -30,18 +32,24 @@ public synchronized Request authenticate(Route route, Response response) throws
         if (response.request().header(AUTHORIZATION_HEADER) == null) {
             throw new IOException("Cannot refresh authentication token due to missing " + AUTHORIZATION_HEADER + " header");
         }
-        //refresh the token
-        LOGGER.log(Level.FINE, "Refreshing OAuth2 Token");
-        OAuth2Token updatedToken = tokenProvider.refreshToken();
-        String accessToken = updatedToken.getAccessToken();
-        if (accessToken == null || accessToken.isEmpty()) {
-            throw new IOException("No access token present in refreshed authentication token");
-        }
-        if (AccessTokenValidator.isTokenExpired(updatedToken)) {
+        OAuth2Token updatedToken;
+        //check if refresh token on current token is still valid
+        String refreshToken = token.getRefreshToken();
+        DecodedJWT jwt = JWT.decode(refreshToken);
+        if (AccessTokenValidator.isTokenExpired(jwt)) {
+            //if expired we need to get a new token
```

Comment:
This case needs a logged message. I think INFO is an ok level to log this at since it would be infrequent, but is meaningful.

Reply:
Added logged message